### PR TITLE
refactor: Remove SER_GETHASH, hard-code client version in CKeyPool serialize

### PR DESCRIPTION
**Pull request by maflcko**
Removes a bunch of redundant, dead or duplicate code.

Uses the idea from and finishes the idea https://github.com/bitcoin/bitcoin/pull/28428 by theuni

### CODE REVIEW

**Comment by theuni** [comment]
Concept ACK.

**Comment by theuni** [comment]
Mind splitting up the last commit? The version number change needs a bit of scrutiny imo, but it's hidden inside an innocuous commit message.

File: src/wallet/scriptpubkeyman.h [hunk 1]
```diff
@@ -123,20 +123,14 @@ class CKeyPool
     template<typename Stream>
     void Serialize(Stream& s) const
     {
-        int nVersion = s.GetVersion();
-        if (!(s.GetType() & SER_GETHASH)) {
-            s << nVersion;
-        }
+        s << int{259900}; // Unused field, writes the highest client version ever written
```
Comment (kevkevinpal) [inline src/wallet/scriptpubkeyman.h:126]: nit: in the comment does "highest client version ever written" mean that this is the highest client version that can be written or that we've ever seen written?

Suggestion
"Unused field, writes the highest possible client version"

feel free to ignore this nit aswell!

**Review by kevkevinpal — APPROVED** [review]
added one nit but otherwise ACK [fac29a0](https://github.com/bitcoin/bitcoin/pull/28508/commits/fac29a0ab19fda457b55d7a0a37c5cd3d9680f82)

**Comment by stickies-v** [comment]
Concept ACK

**Comment by ajtowns** [comment]
ACK fac29a0ab19fda457b55d7a0a37c5cd3d9680f82

**Pull request merged**